examples/http/server.rb

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'rubygems'
+require 'bundler/setup'
+require 'sinatra/base'
+# Require otel-ruby
+require 'opentelemetry/sdk'
+
+SDK = OpenTelemetry::SDK
+OpenTelemetry.tracer_factory = SDK::Trace::TracerFactory.new
+
+exporter = SDK::Trace::Export::ConsoleSpanExporter.new
+processor = SDK::Trace::Export::SimpleSpanProcessor.new(exporter)
+OpenTelemetry.tracer_factory.add_span_processor(processor)
+
+# Rack middleware to extract span context, create child span, and add
+# attributes/events to the span
+class OpenTelemetryMiddleware
+  def initialize(app)
+    @app = app
+    @formatter = formatter = OpenTelemetry.tracer_factory.http_text_format
+    @tracer = OpenTelemetry.tracer_factory.tracer('sinatra', 'semver:1.0')
+  end
+
+  def call(env)
+    # Extract context from request headers
+    context = @formatter.extract(env) { |carrier, key| carrier[key] }
+
+    status, headers, response_body = 200, {}, ''
+
+    # Span name SHOULD be set to route:
+    span_name = env['PATH_INFO']
+
+    # Span kind MUST be `:server` for a HTTP server span
+    @tracer.in_span(span_name, kind: :server, with_parent_context: context) do |span|
+      # For attribute naming, see
+      # https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-semantic-conventions.md#http-server
+      span.set_attribute('component', 'http')
+      span.set_attribute('http.method', env['REQUEST_METHOD'])
+      span.set_attribute('http.route', env['PATH_INFO'])
+      span.set_attribute('http.url', env['REQUEST_URI'])
+      span.add_event(name: 'handle http.server.get')
```

Comment:
This is a slightly odd example. A more realistic one might be the size of the response body (after the `@app.call`).

Reply:
@fbogsany as an `event`?  Okay, I was reading the name event literally as in, this thing happened at this time.

Reply:
That's true, but for this particular operation, that's implicitly the start of the span, so the event isn't that interesting.

I like this blog post as an example of events in a trace (look at the pictures rather than the text): https://medium.com/google-cloud/tracing-google-cloud-8f0c8ba8181c. There, the events are informative little snippets about things that occurred during the span. Some of them may be numeric values (like `146 bytes sent` and `40 bytes received`) while others are things like `DNSStart` and `DNSDone` to record when DNS lookup happened. The latter could conceivably be a span itself, but doesn't have to be.

Reply:
I guess my larger point here is that events should typically be used to add colour to the span, and this specific one doesn't really do that.